custom/abdm/milestone_one/utils/decorators.py

```diff
@@ -8,18 +8,21 @@
 logger = logging.getLogger(__name__)
 
 
-def required_request_params(params: List[str]):
+def required_request_params(params):
     """
-    A version of the requires_privilege decorator which raises an Http404
-    if PermissionDenied is raised.
+    Checks if the parameters provided in the decorator(a list of strings) are present in the DRF request.
```

Comment:
:+1:

Reply:
These are in the POST body of request. Do you mean something else, which I misinterpreted?

Reply:
Okay.
I think its the term parameters that is being used. I would always think of these as params that are a part of the URL.
A better name could be `required_request_data` or `required_request_post_data`. Not sure what a better name could be for params.

Reply:
https://github.com/dimagi/commcare-hq/pull/32332/commits/3ff5dbbc9bf71ad22444349a2cec64bfe2c9b92b